src/graph_input.cpp

```diff
@@ -26,6 +26,20 @@ unsigned long read_ulong(char const* source, char** end)
     return value;
 }
 
+float read_float(char const* source, char** end)
+{
+    float const value = std::strtof(source, end);
+
+#if !defined(__clang__)
```

Comment:
what happens when clang is used? I think a comment to explain this code would be useful.

Reply:
I added a comment concerning the preprocessing pragma checking if the clang compiler is used.